fakestorage/server.go

```diff
@@ -270,7 +271,11 @@ func (s *Server) buildMuxer() {
 
 // publicHostMatcher matches incoming requests against the currently specified server publicHost.
 func (s *Server) publicHostMatcher(r *http.Request, rm *mux.RouteMatch) bool {
-	return r.Host == s.publicHost
+	if strings.Contains(s.publicHost, ":") {
+		return r.Host == s.publicHost
+	}
+	matched, _ := regexp.MatchString("^"+regexp.QuoteMeta(s.publicHost), r.Host)
```

Comment:
doesn't look like the right thing here. Will fix to make sure we don't match hosts partially either (will be annoying to test though).